code/HabitTracker/app/src/main/java/com/example/habittracker/DatabaseManager.java

```diff
@@ -174,6 +182,34 @@ public void onFailure(@NonNull Exception e) {
         return docRef;
     }
 
+    /**
+     * Delete a habit for a user
+     * @param userid        {@code String} User ID
+     * @param habitTitle    {@code String} The title of the habit to be deleted
+     */
+    public void deleteHabitDocument(String userid, String habitTitle) {
+        DocumentReference docRef = usersColRef
+                .document(userid)
+                .collection(habitsColName)
+                .document(habitTitle);
+
+        docRef.delete()
```

Comment:
When you delete a habit, you also have to delete all of the HabitEvents documents for that habit. Deleting the HabitEvents have to be done prior to deleting the habit.
This is because, when the habit is deleted, it's HabitEvents subcollection is not automatically deleted by Firestore.

Reply:
@PaoVal this is very important. Can you make this change now or do you want to address it in the future?

Reply:
That is already happening

Reply:
It is automatically deleted by firestore? As far as I am aware? Deleting the habit deletes everything being contained within the habit, including the habit events.

Reply:
@zarifmahfuz

Reply:
I have tested this. and it works as is currently implemented according tor project specifications

Reply:
As far as I am aware subcollections are not deleted automatically when a parent collection is deleted: https://firebase.google.com/docs/firestore/manage-data/delete-data